oncotext/rationale_net_wrapper.py

```diff
@@ -99,12 +120,36 @@ def label_reports(name, un_reports, config, logger):
             gen, model = model_utils.get_model(args, embeddings, None)
             test_stats  = train_utils.test_model(test_data, model, gen, args)
             preds = test_stats['preds']
+            
         except Exception as e:
             logger.warn("RN Wrapper. {} model failed to label reports! Following Exception({}). Populating all reports with 0 label".format(diagnosis, e))
             preds = np.zeros(len(test_data), dtype=int)
 
-        for i in range(len(test_data)):
-            prediction = label_maps[diagnosis][preds[i]]
-            test_data.dataset[i][diagnosis] = prediction
+        if label_maps[diagnosis][0] == "NUM":
+            try:
```

Comment:
I would recomment refactor this as:
```
for i in range(len(data))):
       text = blah
       pred = helper.get_pred(blah)
```
and moving the details to a helper function/file